doc/source/data/api/input_output.rst

```diff
@@ -1,150 +1,149 @@
-.. _data_api:
 
-Ray Datasets API
-================
+Input/Output
+============
 
-Creating Datasets
------------------
+Synthetic Data
+--------------
 
 .. autofunction:: ray.data.range
+
 .. autofunction:: ray.data.range_table
+
 .. autofunction:: ray.data.range_tensor
+
+Python Objects
+--------------
+
+.. autofunction:: ray.data.from_items
+
+CSV
+---
+
 .. autofunction:: ray.data.read_csv
+
+.. automethod:: ray.data.Dataset.write_csv
+    :noindex:
+
+JSON
+----
+
 .. autofunction:: ray.data.read_json
-.. autofunction:: ray.data.read_parquet
-.. autofunction:: ray.data.read_parquet_bulk
-.. autofunction:: ray.data.read_numpy
-.. autofunction:: ray.data.read_text
-.. autofunction:: ray.data.read_binary_files
-.. autofunction:: ray.data.read_datasource
-.. autofunction:: ray.data.from_items
-.. autofunction:: ray.data.from_arrow
-.. autofunction:: ray.data.from_arrow_refs
-.. autofunction:: ray.data.from_huggingface
-.. autofunction:: ray.data.from_spark
-.. autofunction:: ray.data.from_dask
-.. autofunction:: ray.data.from_modin
-.. autofunction:: ray.data.from_mars
+
+.. automethod:: ray.data.Dataset.write_json
+    :noindex:
+
+Pandas
+------
+
 .. autofunction:: ray.data.from_pandas
+
 .. autofunction:: ray.data.from_pandas_refs
-.. autofunction:: ray.data.from_numpy
-.. autofunction:: ray.data.from_numpy_refs
 
-.. _dataset-api:
+.. automethod:: ray.data.Dataset.to_pandas
+    :noindex:
 
-Dataset API
------------
+.. automethod:: ray.data.Dataset.to_pandas_refs
+    :noindex:
 
-.. autoclass:: ray.data.Dataset
-    :members:
+NumPy
+-----
 
-.. _dataset-pipeline-api:
+.. autofunction:: ray.data.read_numpy
 
-Block API
----------
+.. autofunction:: ray.data.from_numpy
 
-.. autoclass:: ray.data.block.BlockExecStats
-    :members:
+.. autofunction:: ray.data.from_numpy_refs
 
-.. autoclass:: ray.data.block.BlockMetadata
-    :members:
+.. automethod:: ray.data.Dataset.write_numpy
+    :noindex:
 
-.. autoclass:: ray.data.block.BlockAccessor
-    :members:
+.. automethod:: ray.data.Dataset.to_numpy_refs
+    :noindex:
 
-DatasetContext API
-------------------
+Parquet
```

Comment:
Nit: I'd vote for having this as the first file format, before CSV and JSON, since it's our best-supported file format, with the most features (e.g. projection and predicate pushdown) and the most thoroughly tested performance.

Reply:
Updated!